multistream-select/src/protocol/listener.rs

```diff
@@ -112,8 +112,22 @@ impl<R> Sink for Listener<R>
 				}
 			}
 
-			ListenerToDialerMessage::ProtocolsListResponse { list: _list } => {
-				unimplemented!()
+			ListenerToDialerMessage::ProtocolsListResponse { list } => {
+				let mut out_msg = varint::encode(list.len());
```

Comment:
Again, should this be synchronous?

Reply:
`list` is a `Vec<Bytes>` here. There's nothing to be made asynchronous.

Reply:
Great, that's fine then. I probably should have checked that myself